### ISSUE TITLE
[Docs] Need to illustrate the option `Slack` 

### ISSUE THREAD

**Issue by spencercjh** [uber-go/ratelimit#121]

**Comment by aneeskA** [comment]
Yes, please. What is the meaning of `Slack` in a leaky bucket algorithm? I have an observation where when the application do not `Take()` for a period of time, the rate limiting stops to apply on the next `Take()`. I saw a comment that `WithoutSlack` is the way to fix this problem. I want to make an educated decision. Please help with some documentation.

**Comment by spencercjh** [comment]
There was a similar issue about it: #4 .

**Comment by rabbbit** [comment]
We'll be happy to accept PRs that update the documentation.

**Comment by spencercjh** [comment]
A Chinese blog to illustrate `slack` and #119, #120: https://colobu.com/2023/12/05/two-bugs-of-uber-ratelimit/ .

While it partially explains the context of the [bug](#119), it still lacks a direct explanation of `slack` and we are always approaching this issue from an observer's perspective, missing the original author's direct thoughts on this design about `slack`.

If no one can clarify this, please mention that `slack` defaults to 10 and the consequences it brings. Even consider modifying the usage example of this library to guide everyone new to use it with `ratelimit.WithoutSlack`.

If you agree with my comments, I can submit PRs to make contributions.

**Comment by rabbbit** [comment]
Apologies for the delay.

I see the points you're making. If you can put up the PRs, I'll accept them. Please make them small though - one for example, one for the documentation (perhaps in the FAQ)?

Thank you for the Chinese blog post. I'm surprised by the reported performance degradation, I'll try to run some more tests.

**Comment by spencercjh** [comment]
> Apologies for the delay.
> 
> I see the points you're making. If you can put up the PRs, I'll accept them. Please make them small though - one for example, one for the documentation (perhaps in the FAQ)?
> 
> Thank you for the Chinese blog post. I'm surprised by the reported performance degradation, I'll try to run some more tests.

Thanks for your reply. I will try to do some docs works.
